### PR TITLE
Add documentation for feature metrics

### PR DESCRIPTION
**Pull request by aanm**
Add documentation for feature metrics introduced in https://github.com/cilium/cilium/pull/35852

### CODE REVIEW

File: Documentation/observability/metrics.rst [hunk 1]
```diff
@@ -265,6 +265,50 @@ disable the following two metrics as they generate too much data:
...
+  Cluster Mesh.
+
+- **Control Plane** (:ref:`cilium-feature-controlplane`)
+
+  These metrics track control plane configurations, including identity
+  allocation modes and IP address management (IPAM).
+
+- **Datapath** (:ref:`cilium-feature-datapath`)
+
+  Metrics in this group monitor datapath configurations, such as internet
```
Comment (qmonnet) [inline Documentation/observability/metrics.rst]:
```suggestion
  Metrics in this group monitor datapath configurations, such as Internet
```

File: Documentation/observability/metrics.rst [hunk 2]
```diff
@@ -631,6 +675,35 @@ option ``--enable-metrics``.  By default, the operator will expose metrics on
...
+
+- **Advanced Connectivity and Load Balancing** (:ref:`cilium-operator-feature-adv-connect-and-lb`)
+
+  This category includes features related to advanced networking and load
+  balancing capabilities, such as Gateway API, Ingress Controller, LB IPAM,
+  Node IPAM and L7 Aware Traffic Management.
+
+For example, to check if the Gateway API is enabled on a Cilium operator,
+observe the metric ``cilium_operator_feature_adv_connect_and_lb_gateway_api_enabled``.
+All metrics follows the format ``cilium_operator_feature`` + group name + metric name.
```
Comment (qmonnet) [inline Documentation/observability/metrics.rst:694]:
```suggestion
All metrics follow the format ``cilium_operator_feature`` + group name + metric name.
```

File: Documentation/observability/metrics.rst [hunk 3]
```diff
@@ -631,6 +675,35 @@ option ``--enable-metrics``.  By default, the operator will expose metrics on
...
+A value of ``0`` indicates that the feature is disabled, while ``1`` indicates it
+is enabled.
+
+.. note::
+
+   For metrics of type "counter," the operator has processed the associated object
```
Comment (qmonnet) [inline Documentation/observability/metrics.rst:700]:
```suggestion
   For metrics of type "counter", the operator has processed the associated object
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
This is not only a doc change. Would you mind providing a brief description and motivation for the feature-helm-generator and related CI steps that you introduce in your commit title+description, please?

**Review by qmonnet — APPROVED** [review]
Thank you!

**Review by viktor-kurchenko — APPROVED** [review]

**Review by doniacld — APPROVED** [review]

**Pull request merged**